cmd/kube-controller-manager/app/plugins.go

```diff
@@ -43,7 +43,22 @@ func ProbeRecyclableVolumePlugins() []volume.VolumePlugin {
 	// The list of plugins to probe is decided by the kubelet binary, not
 	// by dynamic linking or other "magic".  Plugins will be analyzed and
 	// initialized later.
-	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins()...)
-	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins()...)
+	allPlugins = append(allPlugins, host_path.ProbeVolumePlugins(defaultRecycler(60))...)
+	allPlugins = append(allPlugins, nfs.ProbeVolumePlugins(defaultRecycler(300))...)
 	return allPlugins
 }
+
+// the default recycler:
+//		1. validates the /scrub directory exists
+// 		2. creates a text file to be scrubbed
+//		3. performs rm -rf on the directory
+//		4. tests to see if the directory is empty
+// the pod fails if the error code is returned
+func defaultRecycler(timeoutInSeconds int64) *volume.RecyclableVolumeConfig {
+	return &volume.RecyclableVolumeConfig{
+		ImageName: "gcr.io/google_containers/busybox",
+		Command:   []string{"/bin/sh"},
+		Args:      []string{"-c", "test -e /scrub && echo $(date) > /scrub/trash.txt && rm -rf /scrub/* && test -z \"$(ls -A /scrub)\" || exit 1"},
```

Comment:
I noticed this, too.  This code has to know what kind of volumes are made by the plugins, which is poor.